lib/sles4sap/sdaf_library.pm

```diff
@@ -0,0 +1,520 @@
+
+
+    my $file_path;
+    if ($args{deployment_type} eq 'workload_zone') {
+        my $env_reg_vnet = join('-', $args{env_code}, $args{region_code}, $args{vnet_code});
+        $file_path = "LANDSCAPE/$env_reg_vnet-INFRASTRUCTURE/$env_reg_vnet-INFRASTRUCTURE-$job_id.tfvars";
+    }
+    if ($args{deployment_type} eq 'deployer') {
+        my $env_reg_vnet = join('-', $args{env_code}, $args{region_code}, $args{vnet_code});
+        $file_path = "DEPLOYER/$env_reg_vnet-INFRASTRUCTURE/$env_reg_vnet-INFRASTRUCTURE.tfvars";
+    }
+    if ($args{deployment_type} eq 'library') {
+        my $env_reg = join('-', $args{env_code}, $args{region_code});
+        $file_path = "LIBRARY/$env_reg-SAP_LIBRARY/$env_reg-SAP_LIBRARY.tfvars";
+    }
+    if ($args{deployment_type} eq 'sap_system') {
+        my $env_reg_vnet_sid = join('-', $args{env_code}, $args{region_code}, $args{vnet_code}, $args{sap_sid});
+        $file_path = "SYSTEM/$env_reg_vnet_sid/$env_reg_vnet_sid-$job_id.tfvars";
+    }
+
+    my $result = join('/', deployment_dir(), 'WORKSPACES', $file_path);
+    return $result;
+}
+
+=head2 prepare_sdaf_repo
+
+   prepare_sdaf_repo(
+        [, env_code=>$env_code]
+        [, region_code=>$region_code]
+        [, workload_vnet_code=>$workload_vnet_code]
+        [, deployervnet_code=>$workload_vnet_code]
+        [, sap_sid=>$sap_sid]);
+
+Prepares directory structure and Clones git repository for SDAF samples and automation code.
+
+B<env_code>: Code for SDAF deployment env. Default: 'SDAF_ENV_CODE'
+
+B<deployer_vnet_code>: Deployer virtual network code. Default 'SDAF_DEPLOYER_VNET_CODE'
+
+B<workload_vnet_code>: Virtual network code for workload zone. Default: 'SDAF_WORKLOAD_VNET_CODE'
+
+B<region_code>: SDAF internal code for azure region. Default: 'SDAF_REGION_CODE'
+
+B<sap_sid>: SAP system ID. Default 'SAP_SID'
+
+=cut
+
+sub prepare_sdaf_repo {
+    my (%args) = @_;
+    $args{env_code} //= get_required_var('SDAF_ENV_CODE');
+    $args{deployer_vnet_code} //= get_required_var('SDAF_DEPLOYER_VNET_CODE');
+    $args{workload_vnet_code} //= get_required_var('SDAF_WORKLOAD_VNET_CODE');
+    $args{region_code} //= get_required_var('SDAF_REGION_CODE');
+    $args{sap_sid} //= get_required_var('SAP_SID');
+
+    my $deployment_dir = deployment_dir();
+    my @git_repos = ("https://github.com/Azure/sap-automation.git sap-automation",
+        "https://github.com/Azure/sap-automation-samples.git samples");
+
+    assert_script_run("mkdir -p $deployment_dir; cd \$_");
+    assert_script_run('mkdir -p ' . log_dir());
+
+    foreach (@git_repos) {
+        record_info('Clone repo', "Cloning SDAF repository: $_");
+        assert_script_run("git clone $_ --quiet");
+    }
+
+    assert_script_run("cp -Rp samples/Terraform/WORKSPACES $deployment_dir/WORKSPACES");
+    # Ensure correct directories are in place
+    my %vnet_codes = (
+        workload_zone => $args{workload_vnet_code},
+        sap_system => $args{workload_vnet_code},
```

Comment:
is it fine that `sap_system` is initialized with `workload_vnet_code` like the previous field?

Reply:
Yes. Workload zone provides the virtual network for sap system so it is the same.

Reply:
`Workload zone provides the virtual network for sap system so it is the same` what about adding this as code comment?